src/Disks/ObjectStorages/DiskObjectStorage.cpp

```diff
@@ -75,6 +76,7 @@ DiskObjectStorage::DiskObjectStorage(
     , read_resource_name_from_config(config.getString(config_prefix + ".read_resource", ""))
     , write_resource_name_from_config(config.getString(config_prefix + ".write_resource", ""))
     , metadata_helper(std::make_unique<DiskObjectStorageRemoteMetadataRestoreHelper>(this, ReadSettings{}, WriteSettings{}))
+    , remove_shared_recursive_file_limit(config.getUInt64(config_prefix + ".remove_shared_recursive_file_limit", DEFAULT_REMOVE_SHARED_RECURSIVE_FILE_LIMIT))
```

Comment:
@Algunenano 
It's inside a disc specification in config, I believe we have `Settings[]` only for top-level server settings, don't we?

Reply:
Settings[] is for settings that can be changed per query (with some exceptions I think). This should be a server setting (which sets the default), but as it's added right now it's useless and probably will confuse people, since it won't do anything if you change it in anything else but the config. I'll send a PR to remove the setting for now